yang/frr-pim.yang

```diff
@@ -0,0 +1,253 @@
+      type uint32 {
+        range "1..4294967295";
+      }
+      default 1;
+      description "DR (Designated Router) priority";
+    }
+
+    leaf hello-interval {
+      type uint16 {
+        range "1..180";
+      }
+      default "30";
+      description "Hello interval";
+    }
+
+    leaf hello-holdtime {
+      type uint16 {
+        range "1..180";
+      }
+      description "Hello holdtime";
+    }
+
+    leaf bfd {
+      type boolean;
+      default "false";
+      description
+        "'true' to suppport BFD for the interface.";
+    }
+
+    leaf bsm {
+      type boolean;
+      default "false";
+      description
+        "Enables BSM support on the interface.";
+    }
+
+    leaf unicast-bsm {
+      type boolean;
+      default "false";
+      description
+        "Accept/Send unicast BSM on the interface.";
+    }
+
+    leaf use-source {
+      type inet:ipv4-address;
+      description
+        "Primary address of the interface set by user.";
+    }
+  } // interface-config-attributes
+
+  grouping af-sm-container {
+    description
+      "Grouping of address family SM container.";
+    container pim-sm {
+      description
+        "PIM SM configuration data.";
+      uses sm_common-config-attributes;
+
+      container spt-switchover {
+        description
+          "SPT-Switchover.";
+        leaf spt-action {
+          type enumeration {
+            enum "PIM_SPT_IMMEDIATE" {
+              value 0;
+              description
+                "Immediate switch to SPT Tree.";
+            }
+            enum "PIM_SPT_INFINITY" {
+              value 1;
+              description
+                "Never switch to SPT Tree.";
+            }
+          }
+          default "PIM_SPT_IMMEDIATE";
+          description
+            "SPT-Switchover action";
+        }
+
+        leaf prefix-list {
+          when "../spt-action = 'PIM_SPT_INFINITY'" {
+            description
+              "This leaf is only valid when the spt action
+              is PIM_SPT_INFINITY.";
+          }
+          type string;
+          description
+            "Prefix-List to control which groups to switch.";
+        }
+     }
+
+      container ssm {
+        presence
+          "Present to enable SSM (Source-Specific Multicast).";
+        description
+          "SSM (Source-Specific Multicast) attributes.";
+        leaf range-policy {
+          type string;
+          description
+            "Policy used to define SSM address range.";
+        }
+      } // ssm
+    } // sm
+  } // af-sm-container
+
+  /*
+   * Configuration data nodes
+   */
+
+  augment "/frr-rt:routing/frr-rt:control-plane-protocols/"
+        + "frr-rt:control-plane-protocol" {
+    container pim {
+      when "../frr-rt:type = 'frr-rt:pim'" {
+        description
+          "This container is only valid for the 'pim' routing
+           protocol.";
+      }
+      description
+        "PIM configuration data.";
+
+      list address-family {
```

Comment:
Presently FRR only supports IPv4. Furthermore, it doesn't make sense to include all of the pim attributes on a per AF basis - these are configured globally within the VRF, not by AF

Reply:
Address family was added because if we have PIMv6 then having address-family helps extending existing Yang without impacting IPV4 PIM.